### PR TITLE
Update shard db sections in the example config

### PR DESCRIPTION
**Pull request by undertome**
## High Level Overview of Change

Update the example config file to reflect recent changes to the shard database.

### Context of Change

The example config file has had obsolete keys removed and new sections/keys added.

### Type of Change

- [x ] Documentation Updates

### CODE REVIEW

File: cfg/rippled-example.cfg [hunk 1]
```diff
@@ -1359,7 +1365,14 @@ advisory_delete=0
 # https://xrpl.org/history-sharding.html
 #[shard_db]
 #path=/var/lib/rippled/db/shards/nudb
-#max_size_gb=500
```
Comment (MarkusTeufelberger) [inline cfg/rippled-example.cfg:1362]: Was this setting removed or just renamed?

Comment (undertome) [reply cfg/rippled-example.cfg:1362]: I'd say it was removed since `max_historical_shards` has a different meaning.

**Review by miguelportilla — APPROVED** [review]
👍

File: cfg/rippled-example.cfg (hunk 1 above)
Comment (MarkusTeufelberger) [reply cfg/rippled-example.cfg:1362]: Then having "500" there is going to be unexpectedly large for a lot of people.

Also `max_historical_shards` isn't documented around line 960, only here.

Lastly having a section in the "actual" configuration section yet completely commenting it out seems like a strange choice. If you're only concerned about giving an example on how to configure the `[shard_db]` section, you already have the (nearly) full documentation in line ~950. If this is really that important for the network health, how about uncommenting this section and storing let's say a dozen historic shards by default?

Comment (undertome) [reply cfg/rippled-example.cfg:1362]: Set the commented value for `max_historical_shards` to 50 and added documentation for this optional field. There's already a precedent for providing commented-out sections as starter templates so I'm inclined to leave that as is.

**Review by mDuo13 — APPROVED** [review]

**Pull request closed without merging**